pkg/components/pluggable/registry.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2021 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pluggable
+
+import (
+	"github.com/dapr/dapr/pkg/components"
+	"github.com/dapr/kit/logger"
+)
+
+var (
+	log = logger.NewLogger("pluggable-components")
+	// registries is a map from the given pluggable component type and its register.
+	registries map[components.PluggableType]func(components.Pluggable)
+)
+
+// AddRegistryFor adds a new register function to the puggable registry.
+func AddRegistryFor[T any](cmpType components.PluggableType, regFunc func(componentFactory func(logger.Logger) T, names ...string), factory func(logger.Logger, components.Pluggable) T) {
+	registries[cmpType] = func(pc components.Pluggable) {
+		regFunc(func(l logger.Logger) T {
+			return factory(l, pc)
+		}, pc.Name)
+	}
+}
+
+// Register register all plugable components and returns the total number of registered components.
+func Register(pcs ...components.Pluggable) int {
+	registeredComponents := 0
+	for _, pc := range pcs {
+		register, ok := registries[pc.Type]
+		if !ok {
+			log.Warnf("%s is not registered as a pluggable component", pc.Type)
```

Comment:
Can you log.Info() when the registration is successful? It can help debugging and is not too noisy since is not a per request type of message.

Reply:
I do log how many components were successfully registered [here](https://github.com/dapr/dapr/pull/5102/files#diff-e56f9bb947923c6f8bb4c995bf4b9469a0ed117276223d16b177dcb04399a4bcR638).

Seems good enough to you?

Reply:
The goal is to log each one, not just the count. It is more helpful for debugging in production.